### PR TITLE
Add hashableModels to additional properties

### LINKED ISSUE #9473: [REQ][swift5] Add config flag to control Hashable conformance behavior
### Is your feature request related to a problem? Please describe.

In v5.1.1 https://github.com/OpenAPITools/openapi-generator/pull/9166 introduced a new default behavior that adopts Hashable for every model via synthetization. That way custom implementations won't be called anymore.

## Describe the solution you'd like

After talking with @4brunu we decided that a config flag to control the behavior in general would be good.
On model level you can always use the `x-swift-hashable: true` to enable Hashable conformance.

We should discuss if the feature should be enabled or disabled as default.

## Describe alternatives you've considered

Multiple alternatives would be possible

- Revert https://github.com/OpenAPITools/openapi-generator/pull/9166
- Add documentation for `x-swift-hashable` (I couldn't find it anywhere)

### PR DESCRIPTION
**Pull request by maoyama**
Resolves #9473 , Related #9166 
- Add hashableModels to additional properties (default: true)
- Revert x-swift-hashable

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

### CODE REVIEW

**Comment by 4brunu** [comment]
@maoyama Thanks for opening this PR.
Overall it looks good.

But could you please rename the option `hashableStruct` to `hashableModels`?
Because the models can either be a struct or a class.

**Comment by maoyama** [comment]
@4brunu You’re right.
I rename the option `hashableStruct` to `hashableModels`.

File: modules/openapi-generator/src/main/resources/swift5/modelObject.mustache [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{^objcCompatible}}{{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} {{#useClasses}}final class{{/useClasses}}{{^useClasses}}struct{{/useClasses}} {{classname}}: Codable, Hashable {
+{{^objcCompatible}}{{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} {{#useClasses}}final class{{/useClasses}}{{^useClasses}}struct{{/useClasses}} {{classname}}: Codable{{#vendorExtensions.x-swift-hashable}}, Hashable{{/vendorExtensions.x-swift-hashable}} {
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:1]: @maoyama shouldn't we also check here for `{{#hashableModels}}...{{/hashableModels}}`?

Comment (maoyama) [reply modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:1]: @4brunu We don't have to check here.
I set x-swift-hashable to true when hashableModels is true to reduce the logic in modelObject.mustache. fb2c45f

Comment (4brunu) [reply modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:1]: Nice, good idea! 👍

**Comment by 4brunu** [comment]
Looks good to me, let's wait for the CI to finish

File: modules/openapi-generator/src/main/resources/swift5/modelObject.mustache (hunk 1 above)
Comment (maoyama) [reply modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:1]: Thanks!

**Comment by 4brunu** [comment]
CI is failing, but it's not related to this PR.

**Pull request merged**